### ISSUE TITLE
Unexpected dtype change with empty slice selection

### ISSUE THREAD

**Issue by nekobon** [InvestmentSystems/static-frame#257]
When we select a boolean Series from a Frame using both index and column selector, and when the selected Series is empty, the boolean dtype is converted to float.

Current behavior:

```python
In [18]: f = sf.Frame.from_element(True, index=[1,2,3], columns=['a'])         
In [19]: f # frame with a bool column
Out[19]: 
<Frame>
<Index> a      <<U1>
<Index>
1       True
2       True
3       True
<int64> <bool>

In [20]: target = sf.Series([False, False, False], index=[1,2,3])   
In [21]: target # all False to creating empty selection
Out[21]: 
<Series>
<Index>
1        False
2        False
3        False
<int64>  <bool>

In [22]: f.loc[target]   
Out[22]: 
<Frame>
<Index> a      <<U1>
<Index>
<int64> <bool> # select an empty Frame, dtype is still bool  

In [23]: f.loc[target, 'a']      
Out[23]: 
<Series: a>
<Index>
<int64>     <float64> # select an empty Series, dtype is converted to float
```

Expected behavior:

```python
In [23]: f.loc[target, 'a']      
Out[23]: 
<Series: a>
<Index>
<int64>  <bool> # same dtype as original
```

**Comment by nekobon** [comment]
The original dtype doesn't have to be boolean. String dtype is converted to float64, too.

```python
In [30]: f = sf.Frame.from_element('fourty-two', index=[1,2,3], columns=['a'])   
In [31]: f                        
Out[31]: 
<Frame>
<Index> a          <<U1>
<Index>
1       fourty-two
2       fourty-two
3       fourty-two
<int64> <<U10>

In [32]: f.loc[target, 'a']        
Out[32]: 
<Series: a>
<Index>
<int64>     <float64> # not <U10 anymore

```

**Comment by flexatone** [comment]
Many thanks for isolating this case. Zero-sized `Series` and `Frame` generally require special handling; here, we just had not carried forward the dtype to the returned `Series`. This is fixed and will be part of the 0.6.36 release.

```python
>>> f = sf.Frame.from_element(True, index=[1,2,3], columns=['a'])    
>>> target = sf.Series.from_element(False, index=[1,2,3])   
>>> f.loc[target, 'a']
<Series: a>
<Index>
<int64>     <bool>

>>> f = sf.Frame.from_element('fourty-two', index=[1,2,3], columns=['a'])   
>>> f.loc[target, 'a']
<Series: a>
<Index>
<int64>     <<U10>
```
